src/instructlab/model/backends/backends.py

```diff
@@ -255,26 +255,37 @@ def ensure_server(
         if backend == VLLM:
             # TODO: resolve how the hostname is getting passed around the class and this function
             vllm_server_process = server_process_func(port, background)
-            logger.info(f"Starting a temporary vLLM server at {temp_api_base}")
+            logger.info("Starting a temporary vLLM server at %s", temp_api_base)
             count = 0
             # TODO should this be configurable?
-            vllm_startup_timeout = 60
-            while count < vllm_startup_timeout:
-                if check_api_base(temp_api_base, http_client):
-                    logger.info(f"vLLM engine successfully started at {temp_api_base}")
-                    break
+
+            vllm_startup_max_attempts = 80  # Each call to check_api_base takes >2s + 2s sleep = ~5 mins of wait time
+            start_time_secs = time()
+            while count < vllm_startup_max_attempts:
                 count += 1
                 logger.info(
-                    f"Waiting for the vLLM server to start at {temp_api_base}, this might take a moment... Retries: {count}/{vllm_startup_timeout}"
+                    "Waiting for the vLLM server to start at %s, this might take a moment... Attempt: %s/%s",
+                    temp_api_base,
+                    count,
+                    vllm_startup_max_attempts,
                 )
-                sleep(5)
-
-            if count >= vllm_startup_timeout:
+                if check_api_base(temp_api_base, http_client):
+                    logger.info("vLLM engine successfully started at %s", temp_api_base)
+                    break
+                if count == vllm_startup_max_attempts:
+                    logger.info(
+                        "Gave up waiting for vLLM server to start at %s after %s attempts",
+                        temp_api_base,
+                        vllm_startup_max_attempts,
+                    )
+                else:
+                    sleep(2)
+
+            if count == vllm_startup_max_attempts:
+                duration = round(time() - start_time_secs, 1)
                 shutdown_process(vllm_server_process, 20)
                 # pylint: disable=raise-missing-from
-                raise ServerException(
-                    f"vLLM failed to start up in {vllm_startup_timeout} seconds"
-                )
+                raise ServerException(f"vLLM failed to start up in {duration} seconds")
```

Comment:
Remove based on suggestion above

Reply:
Can you clarify the suggestion?  I should have switched it to not use f"", but not sure what you are suggesting to remove.

Reply:
Instead of doing two checks for `count == vllm_startup_max_attempts:` just move the logic here to the block above (unless there is a reason to keep them seperate)

Reply:
Ah I see now (that suggestion wasn't showing up in the files changed view).  I like the suggestion.  Only dreading the number of rebases I'm going to have to do before this goes in :(

Reply:
Fixed, thanks!